src/slack_api_util.ts

```diff
@@ -168,6 +203,64 @@ export function copyUserInfoToDbMessageEntry(
     userInfo.lastVoterMessageSecsFromEpoch;
 }
 
+export async function fetchSlackChannelMap(): Promise<
+  [Record<string, string>, Record<string, string>]
+> {
+  const response = await slackAPI.get('conversations.list', {
+    params: {
+      limit: 1000,
+      types: 'public_channel,private_channel',
+      token: process.env.SLACK_BOT_ACCESS_TOKEN,
+    },
+  });
+
+  var fw: Record<string, string> = {};
+  var bw: Record<string, string> = {};
+  if (response.data.ok) {
+    for (const info of response.data.channels) {
+      const name = info['name'] as string;
+      const id = info['id'] as string;
+      fw[name] = id;
+      bw[id] = name;
+    }
+  } else {
+    logger.error(
+      `SLACKAPIUTIL.fetchSlackChannelId: Failed to list Slack channels. Error: ${JSON.stringify(
+        response.data
+      )}.`
+    );
+  }
+  return [fw, bw];
+}
+
+export async function fetchSlackChannelId(
```

Comment:
We have a Redis cache to look up a Slack id given a Slack name: RedisApiUtil.slackPodChannelIds